### PR TITLE
Login

### PR DESCRIPTION
**Pull request by rpiaggio**
This PR implements login for observe web.

For the moment, login is required to show anything and a staff role is required too. We can change this in the future, and allow access to a readonly version for non-logged or non-staff users.

A token is still required to access the ODB though, so it's a pending discussion whether we will:
- use an exposed service token for readonly access; or
- allow certain queries in the ODB without a security token; or
- require all users to log into ODB, even to access it readonly.

### CODE REVIEW

**Comment by rpiaggio** [comment]
BTW, for development, observe should now be accessed as `https://local.lucuma.xyz:8081`.

File: modules/web/src/main/scala/observe/ui/Main.scala [hunk 1]
```diff
@@ -5,67 +5,85 @@ package observe.ui
...
+  private val (router, _ /*routerCtl*/ ) =
     RouterWithProps.componentAndCtl(BaseUrl.fromWindowOrigin, Routing.config)
 
-  val mainApp =
+  private val mainApp =
     ScalaFnComponent
-      .withHooks[Unit]
-      .useStateView(RootModel.Initial)
-      .render((_, state) => router(state))
+      .withHooks[Either[Throwable, Option[UserVault]]]
```
Comment (cquiroz) [inline modules/web/src/main/scala/observe/ui/Main.scala:63]: No `Props`?

File: modules/web/src/main/scala/observe/ui/components/Home.scala [hunk 2]
```diff
@@ -85,81 +86,94 @@ object Home {
...
-                    props.rootModel.get.status,
-                    props.rootModel.get.operator,
-                    props.rootModel.zoom(RootModel.conditions)
-                  )
-                )
-              )
+      .render: (props, ctx, observations, demo) =>
+        import ctx.given
+
+        props.rootModel.get.userVault.map(userVault =>
```
Comment (cquiroz) [inline modules/web/src/main/scala/observe/ui/components/Home.scala:92]: Are we going to have side tabs like in explore?

**Review by cquiroz — APPROVED** [review]
LGTM

File: modules/web/src/main/scala/observe/ui/Main.scala (hunk 1 above)
Comment (rpiaggio) [reply modules/web/src/main/scala/observe/ui/Main.scala:63]: It's such a small component that it didn't seem necessary, but we can extract it into another file with a `case class` for props if need be.

File: modules/web/src/main/scala/observe/ui/components/Home.scala (hunk 2 above)
Comment (rpiaggio) [reply modules/web/src/main/scala/observe/ui/components/Home.scala:92]: I hadn't thought of that. Do you think it would be useful?

**Pull request merged**